gitlab/objects.py

```diff
@@ -541,10 +552,41 @@ def unblock(self, **kwargs):
         raise_error_from_response(r, GitlabUnblockError)
         self.state = 'active'
 
+    def __eq__(self, other):
+        if type(other) is type(self):
+            selfdict = self.as_dict()
+            otherdict = self.as_dict()
+            selfdict.pop(u'password', None)
+            otherdict.pop(u'password', None)
+            return selfdict == otherdict
+        return False
+
 
 class UserManager(BaseManager):
     obj_cls = User
 
+    def search(self, query, **kwargs):
+        """Search users.
+
+        Returns a list of matching users.
+        """
+        url = self.obj_cls._url + '?search=' + query
+        return self._custom_list(url, self.obj_cls, **kwargs)
+
+    def get_by_username(self, username, **kwargs):
+        """Get a user by its username.
+
+        Returns a User object or None if the named user does not
+        exist.
+        """
+        url = self.obj_cls._url + '?username=' + username
+        results = self._custom_list(url, self.obj_cls, **kwargs)
+        assert len(results) in (0, 1)
+        try:
+            return results[0]
+        except IndexError:
+            return None
```

Comment:
Other methods raise a `GilabGetError` exception when the resource cannot be found. Could you do the same here, instead of returning `None`?